SimFastTiming/FastTimingCommon/interface/MTDDigitizer.h

```diff
@@ -165,7 +179,12 @@ namespace mtd_digitizer {
     
 
   template<class Traits>
-  void MTDDigitizer<Traits>::beginRun(const edm::EventSetup & es) {
+  void MTDDigitizer<Traits>::beginRun(const edm::EventSetup & es) {    
+    edm::ESHandle<MTDGeometry> geom;
+    if( geomwatcher_.check(es) || geom_ == nullptr ) {
+      es.get<MTDDigiGeometryRecord>().get(geom);
+      geom_ = geom.product();
+    }
```

Comment:
Watcher doesn't gain anything in here either.

Reply:
I removed it

Reply:
The watcher is still here. Or did you mean you removed it in your private branch that is not pushed to GitHub yet?

Reply:
Yes. I'm testing before pushing.